modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -871,134 +769,107 @@ void RenderableFieldlinesSequence::extractTriggerTimesFromFileNames() {
 void RenderableFieldlinesSequence::addStateToSequence(FieldlinesState& state) {
     _states.push_back(state);
     _startTimes.push_back(state.triggerTime());
-    _nStates++;
+    ++_nStates;
 }
 
-bool RenderableFieldlinesSequence::getStatesFromCdfFiles(const std::string& outputFolder)
-{
-    std::string seedFilePath;
-    std::string tracingVar;
-    std::vector<std::string> extraVars;
-    if (!extractCdfInfoFromDictionary(seedFilePath, tracingVar, extraVars)) {
-        return false;
-    }
+bool RenderableFieldlinesSequence::getStatesFromCdfFiles() {
+
+    std::vector<std::string> extraMagVars = extractMagnitudeVarsFromStrings();
 
-    std::vector<glm::vec3> seedPoints;
-    if (!extractSeedPointsFromFile(seedFilePath, seedPoints)) {
+    std::unordered_map<std::string, std::vector<glm::vec3>> seedsPerFiles = 
+        extractSeedPointsFromFiles();
+    if (seedsPerFiles.empty()) {
+        LERROR("No seed files found");
         return false;
     }
 
-    std::vector<std::string> extraMagVars;
-    extractMagnitudeVarsFromStrings(extraVars, extraMagVars);
-
     // Load states into RAM!
     for (const std::string& cdfPath : _sourceFiles) {
         FieldlinesState newState;
         bool isSuccessful = fls::convertCdfToFieldlinesState(
             newState,
             cdfPath,
-            seedPoints,
-            tracingVar,
-            extraVars,
+            seedsPerFiles,
+            _manualTimeOffset,
+            _tracingVariable,
+            _extraVars,
             extraMagVars
         );
 
         if (isSuccessful) {
             addStateToSequence(newState);
-            if (!outputFolder.empty()) {
-                newState.saveStateToOsfls(outputFolder);
+            if (!_outputFolderPath.empty()) {
+                newState.saveStateToOsfls(_outputFolderPath);
             }
         }
     }
     return true;
 }
 
-/*
-* Returns false if it fails to extract mandatory information!
-*/
-bool RenderableFieldlinesSequence::extractCdfInfoFromDictionary(std::string& seedFilePath,
-                                                                std::string& tracingVar,
-                                                      std::vector<std::string>& extraVars)
+std::unordered_map<std::string, std::vector<glm::vec3>>
+RenderableFieldlinesSequence::extractSeedPointsFromFiles() 
```

Comment:
If you pass in the `_seedPointDirectory` as a parameter into this function, it doesn't have to be a member function of the class anymore, simplifying it significantly